### PR TITLE
Update weight formula for session (with new_session taking full block)

### PR DESCRIPTION
**Pull request by thiolliere**
~~updating formula for on_initialize requires new benchmarks, and maybe new API (or every cost of SessionHandler and ShouldEndSession will have to be documented)~~

EDIT: as suggested by shawn new_session take full block to make it easier, if in the future it is wanted we can improve on this.

### CODE REVIEW

**Comment by shawntabrizi** [comment]
@thiolliere `on_initialize -> rotate_session -> new_era` right?

We can make this on_initialize take the full block weight. So the weight function should be an if statement, either 0 or `max_value` mandatory

(I think)

File: frame/session/src/lib.rs [hunk 1]
```diff
@@ -530,9 +539,10 @@ decl_module! {
 		fn on_initialize(n: T::BlockNumber) -> Weight {
 			if T::ShouldEndSession::should_end_session(n) {
 				Self::rotate_session();
+				Weight::max_value()
+			} else {
+				0
```
Comment (thiolliere) [inline frame/session/src/lib.rs:544]: actually this is not zero it involves ShouldEndSession, I'm working on it.

File: frame/session/src/lib.rs [hunk 2]
```diff
@@ -530,9 +539,13 @@ decl_module! {
 		fn on_initialize(n: T::BlockNumber) -> Weight {
 			if T::ShouldEndSession::should_end_session(n) {
 				Self::rotate_session();
+				T::MaximumBlockWeight::get()
+			} else {
+				// NOTE: the non-database part of the weight for `should_end_session(n)` is
+				// included as weight for empty block, the database part is expected to be in
+				// cache.
```
Comment (thiolliere) [inline frame/session/src/lib.rs:546]: reviewer should double-check this statement.

Comment (thiolliere) [reply frame/session/src/lib.rs:546]: related to https://github.com/paritytech/substrate/pull/5722

**Pull request merged**